src/video/vid_riva128.c

```diff
@@ -0,0 +1,1072 @@
+
+    mem_mapping_add(&riva128->mmio_mapping, 0, 0, riva128_mmio_read, riva128_mmio_read_w, riva128_mmio_read_l, riva128_mmio_write, riva128_mmio_write_w, riva128_mmio_write_l,  NULL, MEM_MAPPING_EXTERNAL, riva128);
+    mem_mapping_disable(&riva128->mmio_mapping);
+    mem_mapping_add(&riva128->linear_mapping, 0, 0, svga_read_linear, svga_readw_linear, svga_readl_linear, svga_write_linear, svga_writew_linear, svga_writel_linear,  NULL, MEM_MAPPING_EXTERNAL, &riva128->svga);
+    mem_mapping_disable(&riva128->linear_mapping);
+    mem_mapping_add(&riva128->ramin_mapping, 0, 0, riva128_ramin_read, riva128_ramin_read_w, riva128_ramin_read_l, riva128_ramin_write, riva128_ramin_write_w, riva128_ramin_write_l,  NULL, MEM_MAPPING_EXTERNAL, riva128);
+    mem_mapping_disable(&riva128->ramin_mapping);
+
+    mem_mapping_set_handler(&svga->mapping, svga_read, svga_readw, svga_readl, svga_write, svga_writew, svga_writel);
+
+    io_sethandler(0x03c0, 0x0020, riva128_in, NULL, NULL, riva128_out, NULL, NULL, riva128);
+
+    riva128->card = pci_add_card(PCI_ADD_VIDEO, riva128_pci_read, riva128_pci_write, riva128);
+
+    riva128->pci_regs[0x04] = 0x08;
+    riva128->pci_regs[0x07] = 0x02;
+
+    riva128->pci_regs[0x2c] = 0xd2;
+    riva128->pci_regs[0x2d] = 0x12;
+    riva128->pci_regs[0x2e] = 0x00;
+    riva128->pci_regs[0x2f] = 0x03;
+
+    riva128->pci_regs[0x30] = 0x00;
+    riva128->pci_regs[0x32] = 0x0c;
+    riva128->pci_regs[0x33] = 0x00;
+
+	timer_add(&riva128->mtimer, riva128_mclk_poll, riva128, 0);
+
+    return riva128;
+}
+
+
+static int
+riva128_available(void)
+{
+    return rom_present(BIOS_RIVA128_PATH);
+}
+
+
+void
+riva128_close(void *p)
+{
+    riva128_t *riva128 = (riva128_t *)p;
+	
+    svga_close(&riva128->svga);
+    
+    free(riva128);
+}
+
+
+void
+riva128_speed_changed(void *p)
+{
+    riva128_t *riva128 = (riva128_t *)p;
+    
+    svga_recalctimings(&riva128->svga);
+}
+
+
+void
+riva128_force_redraw(void *p)
+{
+    riva128_t *riva128 = (riva128_t *)p;
+
+    riva128->svga.fullchange = changeframecount;
+}
+
+
+static const device_config_t riva128_config[] =
+{
+        {
+                .name = "memory",
+                .description = "Memory size",
+                .type = CONFIG_SELECTION,
+                .selection =
+                {
+                        {
+                                .description = "1 MB",
+                                .value = 1
+                        },
+                        {
+                                .description = "2 MB",
+                                .value = 2
+                        },
+                        {
+                                .description = "4 MB",
+                                .value = 4
+                        },
+                        {
+                                .description = ""
+                        }
+                },
+                .default_int = 4
+        },
+        {
+                .type = -1
+        }
+};

```

Comment:
yet another indenting style. try sticking with one style. There are linters out there that can do that automatically for you